src/components/navigation/questionNavigation/QuestionStepper.tsx

```diff
@@ -0,0 +1,161 @@
+import { useContext, useEffect, useState } from 'react';
+
+import { ExpandMore } from '@mui/icons-material';
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Box,
+  Divider,
+  Grid,
+  Stack,
+  Typography,
+  useMediaQuery,
+} from '@mui/material';
+
+import { useLocalContext } from '@graasp/apps-query-client';
+
+import { hooks } from '../../../config/queryClient';
+import { QUESTION_BAR_CY } from '../../../config/selectors';
+import theme from '../../../layout/theme';
+import { QuizContext } from '../../context/QuizContext';
+import {
+  computeCorrectness,
+  getAllAppDataByQuestionIdForMemberId,
+  getQuestionById,
+} from '../../context/utilities';
+import { QuestionAppDataData } from '../../types/types';
+import QuestionStep from './QuestionStep';
+import QuestionTitleStepper from './QuestionTitleStepper';
+
+const BG_COLOR = '#FCFCFC';
+const BORDER_COLOR = '#CDCDCD';
+const BORDER_WIDTH = 1;
+
+export const QuestionStepper = () => {
+  const enableAccordion = useMediaQuery(theme.breakpoints.down('md'));
+
+  const { questions, currentIdx, currentQuestion, setCurrentIdx, order } =
+    useContext(QuizContext);
+
+  const { data: appData } = hooks.useAppData();
+  const { memberId } = useLocalContext();
+
+  const [currentNumberOfAttempts, setCurrentNumberOfAttempts] =
+    useState<number>(0);
+  const [isCorrect, setIsCorrect] = useState<boolean>(false);
+  const [isExpanded, setIsExpanded] = useState<boolean>(!enableAccordion);
+
+  const maxAttempts = currentQuestion.data.numberOfAttempts ?? 1;
+
+  useEffect(() => {
+    const responses = getAllAppDataByQuestionIdForMemberId(
+      appData,
+      currentQuestion.data.questionId,
+      memberId
+    );
+
+    const response = responses[responses.length - 1];
+
+    const isCorrect = computeCorrectness(
+      currentQuestion.data,
+      response?.data as QuestionAppDataData
+    );
+
+    setIsCorrect(isCorrect);
+    setCurrentNumberOfAttempts(responses.length);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentQuestion, appData]);
+
+  const renderQuestionStep = (questionId: string, index: number) => {
+    const question = getQuestionById(questions, questionId);
+    if (!question) {
+      console.error('question does not exist');
+      return null;
+    }
+
+    const responses = getAllAppDataByQuestionIdForMemberId(
+      appData,
+      questionId,
+      memberId
+    );
+
+    const response = responses[responses.length - 1];
```

Comment:
What if the array is empty?

Reply:
If the array is empty, response will have the value undefined, which is fine, because computeCorrectness takes data or undefined.